src/_pytest/assertion/rewrite.py

```diff
@@ -118,8 +120,8 @@ def exec_module(self, module):
         if write:
             try:
                 os.mkdir(cache_dir)
-            except OSError:
-                e = sys.exc_info()[1].errno
+            except OSError as exc:
```

Comment:
with python3, cant we use the new exception hierarchy here to match the errors directly?

Reply:
(meant as a followup)

Reply:
Yes, that's a good idea.

Related, Python 3 can also reduce the use of `sys.exc_info()` in some cases. Instead of the annoying tuple, the exception instance itself carries the needed info (`type(e), e, e.__traceback__`).